sampo/utilities/resource_usage.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np
+from collections import defaultdict
+from sortedcontainers import SortedList
+from typing import Iterable
+
+from sampo.schemas.schedule import Schedule
+from sampo.schemas.time import Time
+
+
+def get_total_resources_usage(schedule: Schedule, resources_names: Iterable[str] | None = None) -> dict[str, np.ndarray]:
+    df = schedule.full_schedule_df
+    points = df[['start', 'finish']].to_numpy().copy()
+    points[:, 1] += 1
+    points = SortedList(set(points.flatten()))
+    usage = defaultdict(lambda: np.zeros_like(points))
+
+    is_none = resources_names is None
+    resources_names = set(resources_names) if not is_none else {}
+
+    for swork in schedule.works:
+        start = points.bisect_left(swork.start_time)
+        finish = points.bisect_left(swork.finish_time + 1)
+        for worker in swork.workers:
+            if worker.name in resources_names or is_none:
```

Comment:
If `resource_names` is `None`, you receives `TypeError` here

Reply:
no, because in this case resource_names will be initialized to an empty set